### PR TITLE
pythonPackages.gdown: init at 3.11.1

### PR DESCRIPTION
**Pull request by breakds**
###### Motivation for this change
Add gdown which provides both a CLI tool as well as a python package for downloading artifacts from Google Drive.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -1133,6 +1133,12 @@
     githubId = 7716744;
     name = "Berno Strik";
   };
+  breakds = {
+    email = "breakds@gmail.com";
+    github = "breakds";
+    githubId = 1111035;
+    name = "Break Yang";
+  };
```
Comment (jonringer) [inline maintainers/maintainer-list.nix:1141]: please change the commit message to:

```
maintainers: add <name>
```

File: pkgs/development/python-modules/gdown/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  };
+
+  propagatedBuildInputs = [ filelock requests tqdm setuptools ];
+
+  doCheck = false;
+
+  doInstallCheck = true;
+  installCheckPhase = ''
+    $out/bin/gdown --help > /dev/null
+  '';
```
Comment (jonringer) [inline pkgs/development/python-modules/gdown/default.nix:26]: buildPythonApplication's checkPhase maps to installCheckPhase on derviation, and checkPhase = true; is default for python packages
```suggestion
  checkPhase = ''
    $out/bin/gdown --help > /dev/null
  '';
```

Comment (breakds) [reply pkgs/development/python-modules/gdown/default.nix]: Sure, I made the changes and updated. I kept the `doCheck = false;` there to skip running this module's test.

**Comment by breakds** [comment]
I tried to force push to the branch to update the PR. Found that there is an extra commit on `wesnoth` gets slipped in. Being not familiar enough with the pull request mechanism, how I can get rid of it?

Thanks!

**Comment by jonringer** [comment]
```
git rebase -i HEAD~3

# change pick -> drop on commits you want to drop
# save
git push .. .. --force
```

File: pkgs/development/python-modules/gdown/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1p023812hh7w7d08njjsfn0xzldl4m73yx8p243yb2q49ypjl6nz";
+  };
+
+  propagatedBuildInputs = [ filelock requests tqdm setuptools ];
+
+  doCheck = false;
+
```
Comment (jonringer) [inline pkgs/development/python-modules/gdown/default.nix:22]: This disables the checkPhase, running the tests is already disabled by overriding the default checkPhase, so you're good
```suggestion
```

**Comment by breakds** [comment]
Huh didn't figure it was that easy. Okay I updated the diff the dropped the unnecessary commit. Thanks @jonringer!

**Comment by jonringer** [comment]
According to the readme, this is meant to also be used as an application. Do you mind adding the following to `pkgs/top-level/all-packages.nix`:
```
  gdown = with python3Packages; toPythonApplication gdown;
```
More information on this can be found [here](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#topythonapplication-function)

**Comment by breakds** [comment]
> According to the readme, this is meant to also be used as an application. Do you mind adding the following to `pkgs/top-level/all-packages.nix`:
> 
> ```
>   gdown = with python3Packages; toPythonApplication gdown;
> ```
> 
> More information on this can be found [here](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#topythonapplication-function)

Thanks for the reference. Added it and tested it with `nix-build -A gdown` on my machine.

**Review by jonringer — APPROVED** [review]
diff LGTM
commits LGTM
has tests :+1: 

```
https://github.com/NixOS/nixpkgs/pull/91462
3 packages built:
gdown python27Packages.gdown python37Packages.gdown
```

**Comment by breakds** [comment]
Thanks a lot @jonringer !

**Pull request merged**